File: scripts/windows/MockActuator.java

import com.sun.net.httpserver.HttpServer;
import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpExchange;
import java.io.IOException;
import java.io.OutputStream;
import java.net.InetSocketAddress;

public class MockActuator {
    public static void main(String[] args) throws IOException {
        HttpServer server = HttpServer.create(new InetSocketAddress(8081), 0);
        server.createContext("/actuator/health", new HttpHandler() {
            @Override
            public void handle(HttpExchange t) throws IOException {
                String response = "{\"status\":\"UP\",\"components\":{\"db\":{\"status\":\"UP\"}}}";
                send(t, response);
            }
        });
        server.createContext("/actuator/metrics/http.server.requests", new HttpHandler() {
            @Override
            public void handle(HttpExchange t) throws IOException {
                String response = "{\"name\":\"http.server.requests\",\"measurements\":[{\"statistic\":\"COUNT\",\"value\":120.0},{\"statistic\":\"TOTAL_TIME\",\"value\":5.0}]}";
                send(t, response);
            }
        });
        server.setExecutor(null);
        server.start();
        System.out.println("Mock Actuator running on 8081...");
    }

    private static void send(HttpExchange t, String response) throws IOException {
        t.getResponseHeaders().set("Content-Type", "application/json");
        t.sendResponseHeaders(200, response.length());
        OutputStream os = t.getResponseBody();
        os.write(response.getBytes());
        os.close();
    }
}

File: scripts/windows/mock_actuator.py
from http.server import BaseHTTPRequestHandler, HTTPServer
import json

class MockActuator(BaseHTTPRequestHandler):
    def do_GET(self):
        if self.path == '/actuator/health':
            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(b'{"status":"UP","components":{"db":{"status":"UP"},"diskSpace":{"status":"UP"}}}')
        elif '/actuator/metrics/' in self.path:
            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            # Mock http.server.requests
            self.wfile.write(b'{"name":"http.server.requests","measurements":[{"statistic":"COUNT","value":120.0},{"statistic":"TOTAL_TIME","value":5.0}]}')
        elif self.path == '/actuator/metrics':
             self.send_response(200)
             self.send_header('Content-type', 'application/json')
             self.end_headers()
             self.wfile.write(b'{"names":["http.server.requests","jvm.memory.used"]}')
        else:
            self.send_response(404)
            self.end_headers()

def run():
    server_address = ('', 8081) # Run on 8081 to avoid conflict with Agent (8080)
    httpd = HTTPServer(server_address, MockActuator)
    print("Mock Actuator running on 8081...")
    httpd.serve_forever()

if __name__ == '__main__':
    run()
